code/game/objects/structures/watercloset.dm

```diff
@@ -443,13 +444,12 @@
 	to_chat(usr, "<span class='notice'>You start washing \the [I].</span>")
 
 	busy = 1
-	sleep(40)
+	if(!do_after(user, 40, src))
+		busy = 0
+		to_chat(usr, "<span class='notice'>You stop washing \the [I].</span>")
+		return
 	busy = 0
 
-	if(user.loc != location) return				//User has moved
-	if(!I) return 								//Item's been destroyed while washing
-	if(user.get_active_hand() != I) return		//Person has switched hands or the item in their hands
```

Comment:
This seems like an unnecessary removal.

Reply:
Those're all handled by the do_after(), if the state changes between the start and end of it, it'll fail, hitting the to_chat()/return above

Reply:
^

Reply:
Realized I was thinking of old surgery where you can swap hands, don't mind me.